tests/testthat/test-shinytest2-teal.R

```diff
@@ -0,0 +1,368 @@
+  )
+
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(get_active_data_filters(app, "iris"), "Species")
+  expect_identical(get_active_data_filters(app, "mtcars"), c("cyl", "drat", "gear"))
+  expect_identical(
+    get_active_selection_value(app, "iris", "Species"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "cyl"),
+    c("4", "6")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "drat", is_numeric = TRUE),
+    c(3, 4)
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "gear"),
+    c("3", "4", "5")
+  )
+  app$stop()
+})
+
+test_that("module with a reporter creates the reporter tab", {
+  app_no_reporter <- get_test_app_object(
+    data = simple_teal_data,
+    modules = example_module(label = "Example Module")
+  )
+  app_with_reporter <- get_test_app_object(
+    data = simple_teal_data,
+    modules = report_module(label = "Module with Reporter")
+  )
+
+  teal_tabs <- app_with_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    html_nodes("a")
+  reporter_tabs <- setNames(
+    teal_tabs %>%
+      rvest::html_attr("data-value"),
+    teal_tabs %>%
+      rvest::html_text()
+  )
+  teal_tabs <- app_no_reporter$get_html(selector = "#teal-main_ui-root-active_tab") %>%
+    rvest::read_html() %>%
+    html_nodes("a")
+  non_reporter_tabs <- setNames(
+    teal_tabs %>%
+      rvest::html_attr("data-value"),
+    teal_tabs %>%
+      rvest::html_text()
+  )
+
+  expect_identical(
+    non_reporter_tabs,
+    c("Example Module" = "example_module")
+  )
+  expect_identical(
+    reporter_tabs,
+    c("Module with Reporter" = "module_with_reporter", "Report previewer" = "report_previewer")
+  )
+
+  app_no_reporter$stop()
+  app_with_reporter$stop()
+})
+
+
+test_that("show/hide hamburger works as expected", {
+  app <- get_test_app_object(
+    data = simple_teal_data,
+    modules = example_module()
+  )
+
+  get_class_attributes <- function(app, selector) {
+    element <- app$get_html(selector = selector) |>
+      rvest::read_html() |>
+      html_nodes(selector)
+    list(
+      class = element |>
+        html_attr("class"),
+      style = element |>
+        html_attr("style")
+    )
+  }
+
+  primary_attrs <- get_class_attributes(app, ".teal_primary_col")
+  secondary_attrs <- get_class_attributes(app, ".teal_secondary_col")
+
+  expect_true(grepl("col-sm-9", primary_attrs$class))
+  expect_true(grepl("", secondary_attrs$style))
+
+  app$click(selector = ".btn.action-button.filter_hamburger")
+  app$wait_for_idle(timeout = default_idle_timeout)
+  primary_attrs <- get_class_attributes(app, ".teal_primary_col")
+  secondary_attrs <- get_class_attributes(app, ".teal_secondary_col")
+
+  expect_true(grepl("col-sm-12", primary_attrs$class))
+  expect_true(grepl("display: none;", secondary_attrs$style))
+})
+
+test_that("check the teal app", {
```

Comment:
Please name it precisely and write tests accordingly.
```suggestion
test_that("check the teal app", {
```

Reply:
This was a local test spec. It should be completely removed.